src/pybind/mgr/cephadm/module.py

```diff
@@ -1248,27 +1277,28 @@ def _run_cephadm(self,
 
             self.log.debug('args: %s' % (' '.join(final_args)))
             if self.mode == 'root':
-                if stdin:
-                    self.log.debug('stdin: %s' % stdin)
-                script = 'injected_argv = ' + json.dumps(final_args) + '\n'
-                if stdin:
-                    script += 'injected_stdin = ' + json.dumps(stdin) + '\n'
-                script += self._cephadm
-                python = connr.choose_python()
-                if not python:
-                    raise RuntimeError(
-                        'unable to find python on %s (tried %s in %s)' % (
-                            host, remotes.PYTHONS, remotes.PATH))
                 try:
+                    python = connr.choose_python()
+                    cephadm_dest = '/var/lib/ceph/%s/cephadm/%s' % (
+                        self._cluster_fsid, self.get_mgr_id())
```

Comment:
nack. this needs to also contain the hash of our container

Reply:
why?

Reply:
at this point, cephadm doesn't have a stable CLI. Which means we have to assert that the cephadm binary on the remote host is **exactly** the binary we have in our container. 

Different ways to achieve this, like e.g.

* adding the image hash to the path, or
* adding the sha256 of the binary to the path

(maybe you have another idea?)

At the same time, we have to prevent any races here, thus we should make sure a different cephadm binary is stored at a different path name. (We don't have any locking within mgr/cephadm to prevent concurrent requests)

Reply:
@sebastian-philipp ping:
> why?

Reply:
did you saw my reply? 

![image](https://user-images.githubusercontent.com/2574405/96477671-12a16380-1237-11eb-902d-0665ba45fae5.png)